pkg/endpoint/policy.go

```diff
@@ -498,6 +499,230 @@ func (e *Endpoint) IngressOrEgressIsEnforced() bool {
+		// FIXME check for ports names here if it doesn't exist, the throw initial error
+		return 0, err
+	}
+	return uint16(i), nil
+}
+
+func (e *Endpoint) regeneratePolicyNew(owner Owner, opts models.ConfigurationMap) (*policy.Policy, *policy.Policy, uint64) {
+	policyRepo := owner.GetPolicyRepository()
+
+	oldIngressPolicy := e.IngressPolicy
+	oldEgressPolicy := e.EgressPolicy
+	oldPolicyRev := e.policyRevision
+
+	ingressPolicy := policy.NewPolicy()
+	egressPolicy := policy.NewPolicy()
+
+	ctx := &policy.SearchContext{
+		To: e.SecurityIdentity.Labels.LabelArray(),
+	}
+
+	policyRepo.Mutex.RLock()
+	defer policyRepo.Mutex.RUnlock()
+	ing, egress, rev := policyRepo.ResolvePolicy(ctx)
+
+	labelsMap, _ := getLabelsMap()
+
+	// Create ingress policy
+	for _, ig := range ing {
+		// FIXME needs validation of a singe peer on each ingress rule
+		// Accept only FromEntities or FromCIDR or FromEndpoints or ...
+
+		// Create the L4Port first
+		l4Ports := policy.NewL4RuleContexts()
+		for _, toPort := range ig.ToPorts {
+			for _, pp := range toPort.Ports {
+				port, err := e.GetPortNumber(pp.Port)
+				if err != nil {
+					log.Debugf("Error while getting port number for %s", port)
+					continue
+				}
+				l4RuleCtx := policy.L4RuleContext{
+					Port: port,
+					// FIXME do the translation of protocols to uint8
+					// For now 0 means any
+					Proto: 0,
+				}
+				if _, ok := l4Ports[l4RuleCtx]; !ok {
+					l4Ports[l4RuleCtx] = policy.L7RuleContext{}
+				}
+			}
+			// FIXME When/How to store the ProxyPort?
+		}
+
+		// Create the policy ByNumericIdentity
+		for srcID, srcLabels := range *labelsMap {
+			for _, es := range ig.FromEndpoints {
+				// FIXME do we still need to use the policy repo to know if something is Accepted or Deny?
+				if es.Matches(srcLabels) {
+					if _, ok := ingressPolicy.ByNumericIdentity[srcID]; !ok {
+						ingressPolicy.ByNumericIdentity[srcID] = l4Ports
+					}
+				}
+			}
+		}
+
+		// Create the policy ByEntity
+		for _, entity := range ig.FromEntities {
+			ingressPolicy.ByEntity[entity] = l4Ports
+		}
+
+		// Create the policy ByCIDR
+		for _, cidr := range ig.FromCIDR {
+			ingressPolicy.ByCIDR[cidr] = l4Ports
+		}
+	}
+
+	// Create egress policy
+	for _, eg := range egress {
+		// FIXME needs validation of a singe peer on each ingress rule
+		// Accept only ToEntities or ToCIDR or ToService or ...
+
+		// Create the L4Port first
+		l4Ports := policy.NewL4RuleContexts()
+		for _, toPort := range eg.ToPorts {
+			for _, pp := range toPort.Ports {
+				port, err := e.GetPortNumber(pp.Port)
+				if err != nil {
+					log.Debugf("Error while getting port number for %s", port)
+					continue
+				}
+				l4RuleCtx := policy.L4RuleContext{
+					Port: port,
+					// FIXME do the translation of protocols to uint8
+					// For now 0 means any
+					Proto: 0,
+				}
+				if _, ok := l4Ports[l4RuleCtx]; !ok {
+					l4Ports[l4RuleCtx] = policy.L7RuleContext{}
+				}
+			}
+			// FIXME When/How to store the ProxyPort?
```

Comment:
Should be here, shouldn't it? the resulting `policy.Policy` should have all the relevant information to generate BPF map contents/code?

Reply:
You now are mixing datapath details with policy details, I was trying to avoid that :-(

Reply:
True, disregard my comment above.